### PR TITLE
Allow Payments History to sort when sort option is chosen

### LINKED ISSUE #361: Allow Payments History to sort by Title, Reason, or Date when sort option is chosen
Fixed by #366

### PR DESCRIPTION
**Pull request by jgreben**
Fixes #361 
Before:

Screen Shot 2019-07-30 at 5 08 17 PM

After:

Screen Shot 2019-07-30 at 5 08 27 PM

### CODE REVIEW

File: app/views/fines/_payment.html.erb [hunk 1]
```diff
@@ -1,20 +1,20 @@
 <li class="list-group-item">
   <div class="row d-flex flex-wrap position-relative">
     <div class="d-flex flex-row flex-grow-1 justify-content-start w-100 row col-md-12 col-lg-6">
-      <div class="col-4 col-md-4 order-md-first description">
+      <div class="col-4 col-md-4 order-md-first bill_description" data-sort-bill_description="<%= payment.bill_description %>" >
```
Comment (cbeer) [inline app/views/fines/_payment.html.erb]: Can we do something like we're doing on the other models (using a `sort_key` method that knows how to sort + do fallbacks to other columns)?

**Comment by jgreben** [comment]
We are currently not defining any secondary sorting on Reason/Date paid/Title. If we need to do that, let's create another ticket.

File: app/views/checkouts/index.html.erb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<h1 class="sr-only">Checkouts</h1>
+[<h1 class="sr-only">Checkouts</h1>
```
Comment (cbeer) [inline app/views/checkouts/index.html.erb]:
```suggestion
<h1 class="sr-only">Checkouts</h1>
```

File: app/views/fines/_payment.html.erb [hunk 3]
```diff
@@ -1,20 +1,20 @@
...
 
-      <div class="col-5 col-md-4 text-right text-md-left due_date" data-date="<%= payment.payment_date.strftime('%FT%T') %>">
+      <div class="col-5 col-md-4 text-right text-md-left payment_date" data-date="<%= payment.payment_date.strftime('%FT%T') %>" data-sort-payment_date="<%= payment.payment_date %>" >
         <%= l(payment.payment_date, format: :short) %>
       </div>
     </div>
-    <div class="d-flex flex-grow-1 flex-column flex-md-row w-75 row col-md-12 col-lg-6">
-      <h3 class="col-md-10 clamp-3 record-title list-item-title"><%= payment.item_title %></h3>
+    <div class="d-flex flex-grow-1 flex-column flex-md-row w-75 row col-md-12 col-lg-6 item_title" data-sort-item_title="<%= payment.item_title %>">
+      <h3 class="col-md-10 clamp-2 record-title list-item-title"><%= payment.item_title %></h3>
```
Comment (cbeer) [inline app/views/fines/_payment.html.erb]: This looks like maybe a rebase error?

File: app/views/fines/index.html.erb [hunk 4]
```diff
@@ -56,25 +56,25 @@
...
       </div>
     </div>
 
     <div class="d-none d-md-flex row font-weight-bold list-header">
       <div class="row col-md-12 col-lg-6">
-        <div class="col-md-4" data-sort="reason">Reason</div>
-        <div class="col-md-4" data-sort="amount">Amount</div>
-        <div class="col-md-4 active" data-sort="due_date">Paid on</div>
+        <div class="col-md-4" data-sort="bill_description">Reason</div>
+        <div class="col-md-4">Amount</div>
```
Comment (cbeer) [inline app/views/fines/index.html.erb]: I think we'd want to make amount sortable too? It'd be a little weird for one column header not to be, right @jvine ?

File: app/views/fines/_payment.html.erb (hunk 1 above)
Comment (jgreben) [reply app/views/fines/_payment.html.erb]: @jvine what do you think of these secondary sort options? The first one is the chosen sort option and the ones after the secondary:

```
[**payment_date**, item_title, bill_description]
[**item_title**, payment_date, bill_description]
[**bill_amount**, payment_date, item_title, bill_description]
[**bill_description**, payment_date, item_title]
```

**Pull request merged**